fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java

```diff
@@ -381,40 +352,31 @@ default void replayDropConstraint(String name) {
     }
 
     default void dropConstraint(String name, boolean replay) {
-        writeLock();
-        try {
-            Map<String, Constraint> constraintMap = getConstraintsMapUnsafe();
-            if (!constraintMap.containsKey(name)) {
-                throw new AnalysisException(
-                        String.format("Unknown constraint %s on table %s.", name, this.getName()));
-            }
-            Constraint constraint = constraintMap.get(name);
-            constraintMap.remove(name);
-            if (constraint instanceof PrimaryKeyConstraint) {
-                ((PrimaryKeyConstraint) constraint).getForeignTables()
-                        .forEach(t -> t.dropFKReferringPK(this, (PrimaryKeyConstraint) constraint));
-            }
-            if (!replay) {
-                Env.getCurrentEnv().getEditLog().logDropConstraint(new AlterConstraintLog(constraint, this));
-            }
-        } finally {
-            writeUnlock();
+        Map<String, Constraint> constraintMap = getConstraintsMapUnsafe();
+        if (!constraintMap.containsKey(name)) {
+            throw new AnalysisException(
+                    String.format("Unknown constraint %s on table %s.", name, this.getName()));
+        }
+        Constraint constraint = constraintMap.get(name);
+        constraintMap.remove(name);
+        if (constraint instanceof PrimaryKeyConstraint) {
+            ((PrimaryKeyConstraint) constraint).getForeignTables()
+                    .forEach(t -> t.dropFKReferringPK(this, (PrimaryKeyConstraint) constraint));
+        }
+        if (!replay) {
+            Env.getCurrentEnv().getEditLog().logDropConstraint(new AlterConstraintLog(constraint, this));
         }
     }
 
     default void dropFKReferringPK(TableIf table, PrimaryKeyConstraint constraint) {
-        writeLock();
-        try {
-            Map<String, Constraint> constraintMap = getConstraintsMapUnsafe();
-            Set<String> fkName = constraintMap.entrySet().stream()
-                    .filter(e -> e.getValue() instanceof ForeignKeyConstraint
-                    && ((ForeignKeyConstraint) e.getValue()).isReferringPK(table, constraint))
-                    .map(Entry::getKey)
-                    .collect(Collectors.toSet());
-            fkName.forEach(constraintMap::remove);
-        } finally {
-            writeUnlock();
-        }
+        Map<String, Constraint> constraintMap = getConstraintsMapUnsafe();
+        Set<String> fkName = constraintMap.entrySet().stream()
+                .filter(e -> e.getValue() instanceof ForeignKeyConstraint
+                        && ((ForeignKeyConstraint) e.getValue()).isReferringPK(table, constraint))
+                .map(Entry::getKey)
+                .collect(Collectors.toSet());
+        fkName.forEach(constraintMap::remove);
```

Comment:
no lock will see the state of partial update?

Reply:
already locked outside